unicodetools/src/main/java/org/unicode/text/UCD/TestUnicodeInvariants.java

```diff
@@ -691,26 +690,69 @@ private static void inLine(
     }
 
     // A one-token lookahead.
-    // Tokens are defined as runs of [^\p{Pattern_White_Space}\p{Pattern_Syntax}],
-    // or single code points in \p{Pattern_Syntax}.
+    // Tokens are defined as:
+    // 1. words: runs of [^\p{Pattern_White_Space}\p{Pattern_Syntax}];
+    // 2. simple operators: sequences of the form \p{Pattern_Syntax} \p{Mn}*;
+    // 3. explicitly expected sequences of words and simple operators without intervening spaces;
+    //    this allows for contextually accepting operators such as :=, >>, ’s, or .GT.,
+    //    without treating, e.g., every >> as atomic.
     private static class Lookahead {
         // Advances pp through any pattern white space, then looks ahead one token.
         public static Lookahead oneToken(ParsePosition pp, String text) {
             scan(PATTERN_WHITE_SPACE, text, pp, true);
             return oneTokenNoSpace(pp, text);
         }
 
+        // Advances pp through any pattern white space, then looks ahead one token,
+        // treating the given sequences as single tokens.
+        public static Lookahead oneToken(ParsePosition pp, String text, String... sequences) {
+            scan(PATTERN_WHITE_SPACE, text, pp, true);
+            Lookahead result = oneTokenNoSpace(pp, text);
+            if (result == null) {
+                return result;
+            }
+            Lookahead candidate = result;
+            for (; ; ) {
+                boolean candidateIsSequencePrefix = false;
+                for (String s : sequences) {
+                    if (s.startsWith(candidate.token)) {
+                        candidateIsSequencePrefix = true;
+                        break;
+                    }
+                }
+                if (!candidateIsSequencePrefix) {
+                    break;
+                }
+                final Lookahead continuation = oneTokenNoSpace(candidate.next, text);
+                if (continuation == null) {
+                    break;
+                }
+                candidate =
+                        new Lookahead(candidate.token + continuation.token, pp, continuation.next);
+
+                for (String s : sequences) {
```

Comment:
Nit: If you passed the sequences in as some kind of a Collection, then you could replace this loop with .contains(candidate). (And the previous inner loop could probably use a Stream...anyMatch() construct.)